### PR TITLE
fix(metrics): RHICOMPL-1763 s/Counter/Gauge/ for systems_by_os

### PR DESCRIPTION
**Pull request by akofink**
See https://prometheus.io/docs/concepts/metric_types/#counter

Signed-off-by: Andrew Kofink <akofink@redhat.com>

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
I thought a Gauge can't have multiple sub-values, that's why I went with a Counter :see_no_evil:

**Review by skateman — APPROVED** [review]
The Seal of Approval

File: lib/prometheus/business_collector.rb [hunk 1]
```diff
@@ -38,10 +38,10 @@ def initialize
     @client_systems = PrometheusExporter::Metric::Gauge.new(
       'client_systems', 'Systems from clients (excludes Red Hat)'
     )
-    @total_systems_by_os = PrometheusExporter::Metric::Counter.new(
+    @total_systems_by_os = PrometheusExporter::Metric::Gauge.new(
```
Comment (vkrizan) [inline lib/prometheus/business_collector.rb:41]: Gauge is more appropriate as the value can decrease.

See note:
```
Do not use a counter to expose a value that can decrease. For example, do not use a counter for the number of currently running processes; instead use a gauge.
```

**Review by vkrizan — APPROVED** [review]
:+1: 
(:facepalm:, I read the changes the other way, ignore the comment before)

**Pull request merged**